backend/src/test/java/de/wirvsvirus/trackyourbed/HospitalServiceTest.java

```diff
@@ -0,0 +1,318 @@
+  }
+
+  @Test
+  @DisplayName("Should call dependencies and return expected result when called.")
+  void shouldCreateNewHospital() {
+    // GIVEN
+    final String name = "hospital";
+    final int maxCapacity = 23;
+    final String lon = "11.11";
+    final String lan = "12.11";
+    final CreateNewHospital request = new CreateNewHospital()
+        .setMaxCapacity(maxCapacity)
+        .setLon(lon)
+        .setLat(lan)
+        .setName(name);
+    final Hospital hospital = new Hospital();
+    final HospitalDto expected = new HospitalDto();
+
+    when(createNewHospitalMapper.dtoToEntity(request)).thenReturn(hospital);
+    when(hospitalRepository.save(any(Hospital.class))).thenReturn(hospital);
+    when(hospitalDtoMapper.entityToDto(any(Hospital.class))).thenReturn(expected);
+
+    // WHEN
+    final HospitalDto actual = hospitalService.createHospital(request);
+
+    // THEN
+    assertSame(expected, actual);
+    verify(createNewHospitalMapper).dtoToEntity(argThat(r -> {
+      assertSame(request, r);
+      return true;
+    }));
+    verify(hospitalRepository).save(argThat(b -> {
+      assertSame(hospital, b);
+      return true;
+    }));
+    verify(hospitalDtoMapper).entityToDto(argThat(b -> {
+      assertSame(hospital, b);
+      return true;
+    }));
+  }
+
+  @Test
+  void shouldDeleteHospital() {
+    //Given
+    doNothing().when(hospitalRepository).deleteById(any(UUID.class));
+
+    //When
+    hospitalService.deleteHospital(UUID.randomUUID());
+
+    //then
+    verify(hospitalRepository, times(1)).deleteById(any(UUID.class));
+  }
+
+  @Test
+  void shouldGetHospitalById() {
+    //Given
+    final Hospital hospital = getMockHospital();
+    final HospitalDto expected = new HospitalDto();
+    UUID hospitalId = UUID.randomUUID();
+
+    when(hospitalRepository.findById(hospitalId)).thenReturn(Optional.of(hospital));
+    when(hospitalDtoMapper.entityToDto(hospital)).thenReturn(expected);
+
+    //When
+    HospitalDto actual = hospitalService.getHospitalById(hospitalId);
+
+    //then
+    assertSame(expected, actual);
+    verify(hospitalRepository, times(1)).findById(hospitalId);
+    verify(hospitalDtoMapper, times(1)).entityToDto(hospital);
+  }
+
+  @Test
+  void shouldThrowNoSuchHospitalExceptionWhenGetHospitalById() {
+    //Given
+    final Hospital hospital = getMockHospital();
+    UUID hospitalId = UUID.randomUUID();
+
+    when(hospitalRepository.findById(hospitalId)).thenReturn(Optional.empty());
+
+    //When
+    try {
+      hospitalService.getHospitalById(hospitalId);
+    } catch (NoSuchHospitalException ex) {
+      assertEquals("Could not find hospital with ID " + hospitalId, ex.getMessage());
+    }
+
+    //then
+    verify(hospitalRepository, times(1)).findById(hospitalId);
+    verify(hospitalDtoMapper, times(0)).entityToDto(hospital);
+  }
+
+  @Test
+  void shouldGetAllHospitals() {
+    //Given
+    final Hospital hospital1 = new Hospital();
+    hospital1.setName("hos1");
```

Comment:
I would suggest extracting `"hos1"` and `"hos2"` in variables and constructing a `List` of those two varialbes (can be used to verify list size and in `containsInAnyOrder(...)`, for the latter, `toArray()` must be used).